test/bridges/base-bridge-receiver-test.ts

```diff
@@ -0,0 +1,452 @@
+    } = await makeBridgeReceiver({ initialize: false });
+
+    const tx = await wait(baseBridgeReceiver.initialize(govTimelock.address, localTimelock.address));
+
+    expect(await baseBridgeReceiver.govTimelock()).to.eq(govTimelock.address);
+    expect(await baseBridgeReceiver.localTimelock()).to.eq(localTimelock.address);
+    expect(await baseBridgeReceiver.initialized()).to.eq(true);
+
+    expect(event(tx, 0)).to.be.deep.equal({
+      Initialized: {
+        govTimelock: govTimelock.address,
+        localTimelock: localTimelock.address
+      }
+    });
+  });
+
+  it('cannot be reinitialized', async () => {
+    const {
+      baseBridgeReceiver,
+      govTimelock,
+      localTimelock
+    } = await makeBridgeReceiver({initialize: true});
+
+    await expect(
+      baseBridgeReceiver.initialize(govTimelock.address, localTimelock.address)
+    ).to.be.revertedWith("custom error 'AlreadyInitialized()'");
+  });
+
+  it('acceptLocalTimelockAdmin reverts if not called by localTimelock', async () => {
+    const { baseBridgeReceiver } = await makeBridgeReceiver();
+
+    await expect(
+      baseBridgeReceiver.acceptLocalTimelockAdmin()
+    ).to.be.revertedWith("custom error 'Unauthorized()'");
+  });
+
+  it('setLocalTimelock > reverts for unuauthorized caller', async () => {
+    const {
+      baseBridgeReceiver,
+      localTimelock
+    } = await makeBridgeReceiver();
+
+    await expect(
+      baseBridgeReceiver.setLocalTimelock(localTimelock.address)
+    ).to.be.revertedWith("custom error 'Unauthorized()'");
+  });
+
+  it('setLocalTimelock > sets new timelock', async () => {
+    const {
+      baseBridgeReceiver,
+      govTimelock,
+      signers
+    } = await makeBridgeReceiver({ initialize: false });
+
+    const [localTimelockSigner, newLocalTimelockSigner] = signers;
+
+    await baseBridgeReceiver.initialize(
+      govTimelock.address,
+      localTimelockSigner.address
+    );
+
+    const tx = await wait(
+      baseBridgeReceiver.connect(localTimelockSigner).setLocalTimelock(
+        newLocalTimelockSigner.address
+      )
+    );
+
+    expect(await baseBridgeReceiver.localTimelock()).to.eq(newLocalTimelockSigner.address);
+
+    expect(event(tx, 0)).to.be.deep.equal({
+      NewLocalTimelock: {
+        newLocalTimelock: newLocalTimelockSigner.address,
+        oldLocalTimelock: localTimelockSigner.address
+      }
+    });
+  });
+
+  it('setGovTimelock > reverts for unauthorized caller', async () => {
+    const {
+      baseBridgeReceiver,
+      govTimelock
+    } = await makeBridgeReceiver();
+
+    await expect(
+      baseBridgeReceiver.setGovTimelock(govTimelock.address)
+    ).to.be.revertedWith("custom error 'Unauthorized()'");
+  });
+
+  it('setGovTimelock > sets gov timelock', async () => {
+    const {
+      baseBridgeReceiver,
+      govTimelock,
+      signers
+    } = await makeBridgeReceiver({ initialize: false });
+
+    const [localTimelockSigner, newGovTimelockSigner] = signers;
```

Comment:
`govTimelcok` and `localTimelock` are also derived from signers, so probably safer to use some other hard-coded address for the `newGovTimelock`

Reply:
Or maybe you could do it the other way around and use some hard-coded address for the original `govTimelock` and `localTimelock` instead.

Reply:
Disregard this. I misread the setup code